en/docs/devops-and-ci-cd/configure-vpns-on-the-choreo-cloud-data-plane.md

```diff
@@ -0,0 +1,210 @@
+
+    | **Field**                | **Value**                          |
+    |--------------------------|------------------------------------|
+    | **Project Display Name** | Tailscale Proxy Project            |
+    | **Name**                 | tailscale-proxy-project            |
+    | **Project Description**  | My Tailscale project               |
+
+4. Click **Create**. This creates the project and takes you to the project home page.
+
+#### Step 1.2: Create the Tailscale proxy component
+
+1. On the project home page, click **Service** under **Create a Component**.
+2. Enter a display name, component name, and a description for the service. For this guide, let's enter the following values:
+
+    |Field                 |     Value                |
+    |----------------------|--------------------------|
+    |Component Display Name| Tailscale proxy          |
+    |Component Name        | tailscale-proxy          |
+    |Description           | Tailscale proxy component|
+
+3. Click the **Container Registry** tab.
+4. In the **Container Registry** list, select **Choreo Samples Registry**.
+5. Click the **Tailscale Proxy** card.
+6. Click **Create**.
+
+Now you have successfully created the Tailscale proxy. You can proceed to configure and deploy it.
+
+### Step 2: Configure and deploy the Tailscale proxy
+
+Here, you will add the required volume mounts, set the Tailscale authentication key, configure the TCP forwarder, configure endpoints, and deploy the Tailscale proxy. Follow the steps given below:
+
+#### Step 2.1: Add required volume mounts
+
+Tailscale requires the following volume mounts for its operations:
+
+ - `/var/run/tailscale`
+ - `/.local`
+
+To create the volume mounts, follow the step-by-step instructions in [Configure Storage](../devops-and-ci-cd/configure-storage.md). 
+
+#### Step 2.2: Configure and deploy the component
+
+To configure and deploy the component, follow the steps given below:
+
+1. In the left navigation menu, click **Deploy**.
+2. On the **Build Area** card, click **Configure &  Deploy**.
+3. In the **Environment Configurations** pane that opens, click **+ Add** and add the `TS_AUTH_KEY` environment variable as a secret. To add the environment variable, you must obtain an authentication key from your Tailscale network.
+    - For details on how to obtain an authentication key from your Tailscale network, see [Auth keys](https://tailscale.com/kb/1085/auth-keys) in the Tailscale documentation.
+4. Click **Next**.
+5. In the **File Mount** pane that opens, click **+ Add**.
+6. To mount a configuration file to the Tailscale proxy component and specify the port mapping for the TCP forward proxy running there, do the following:
+    1. Specify `/Config.yaml` as the **Mount Path**.
+    2. Specify the following in the sample configuration file:
+       ```
+       portMappings:
+           8080: "100.108.78.93:8090"
+           8081: "100.108.78.93:1433"
+       ```
```

Comment:
Shall we add a note (or something) after this portMappings configuration saying the following?
- In this example configuration, the TCP traffic coming via port `8080` of your tailscale-proxy will be forwarded to the port `8090` of the node with IP `100.108.78.93` in your Tailscale network and the port `8081` maps to the corresponding address. You may find this IP address via [Tailscale machines](https://login.tailscale.com/admin/machines) page of your Tailscale network's admin console or via the Tailscale clients running on your machine.

Reply:
+1.
Added a note.